### PR TITLE
Cirlcleci 2.0

### PR DESCRIPTION
**Pull request by Xorima**
## Description
Use orb 2.0

### Issues Resolved

[List any existing issues this PR resolves]

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: kitchen.dokken.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    driver:
+      image: dokken/centos-7
+      pid_one_command: /usr/lib/systemd/systemd
+
+  - name: fedora-26
+    driver:
+      image: dokken/fedora-26
+      pid_one_command: /usr/lib/systemd/systemd
+
+  - name: fedora-27
```
Comment (damacus) [inline kitchen.dokken.yml]: Please delete me

File: kitchen.dokken.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    driver:
+      image: dokken/centos-6
+      pid_one_command: /sbin/init
+
+  - name: centos-7
+    driver:
+      image: dokken/centos-7
+      pid_one_command: /usr/lib/systemd/systemd
+
+  - name: fedora-26
```
Comment (damacus) [inline kitchen.dokken.yml]: please delete me

File: kitchen.dokken.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    driver:
+      image: dokken/fedora-27
+      pid_one_command: /usr/lib/systemd/systemd
+
+  - name: fedora-latest
+    driver:
+      image: dokken/fedora-latest
+      pid_one_command: /usr/lib/systemd/systemd
+
+  - name: ubuntu-14.04
```
Comment (damacus) [inline kitchen.dokken.yml]: Wut wut

File: kitchen.dokken.yml (hunk 1 above)
Comment (Xorima) [reply kitchen.dokken.yml]: removed all of fedora :D, wait till i add you as a reviewer haha

File: kitchen.yml [hunk 4]
```diff
@@ -28,5 +29,5 @@ suites:
    run_list:
    - recipe[test::default]
    attributes:
-      oink_code: <%= ENV['SNORT_LICENCE'] %>
-      community_rules_url: <%= ENV['COMMUNITY_URL'] %>
+      # oink_code: <%= ENV['SNORT_LICENCE'] %>
+      # community_rules_url: <%= ENV['COMMUNITY_URL'] %>
```
Comment (damacus) [inline kitchen.yml]: I put this in so I can pull the OpenSource license code from my env. I need to think how to do this again 🤔

Comment (Xorima) [reply kitchen.yml]: circleci context could work?

**Comment by Xorima** [comment]
So the problem here is it no longer compiles on debian, and the commands do not seem to throw correctly so it is invisible to us, @damacus what are your thoughts here? I'm tempted to drop debian but that feels like a hack

**Comment by simonwgill** [comment]
> So the problem here is it no longer compiles on debian, and the commands do not seem to throw correctly so it is invisible to us, @damacus what are your thoughts here? I'm tempted to drop debian but that feels like a hack

The compile step failure is because the open-appid feature needs LuaJIT support which is not available. That probably should be disabled for now (configure --disable-open-appid), and the tests pass in that case.

In the long run, using the shortcut format for the execute compile "configure && make && make install && ldconfig" does allow for failures in that process to stop the cookbook. 

On a side note, the ln and copy commands probably should be resources in their own right instead of execute to allow the cookbook to recover certain changes. Interestingly, the presence of the source archive is used to determine if the source should be compiled rather than a signal from the environment.

**Comment by Xorima** [comment]
@simonwgill Can you issue a pr into this branch so we can get this working, unfortunately my knowledge of compiling is non-existent so your help would be greatly appreciated!

**Comment by Xorima** [comment]
@PowerSchill any chance you can resolve that last error on ubuntu please? :)

**Comment by PowerSchill** [comment]
> @PowerSchill any chance you can resolve that last error on ubuntu please? :)

How do you like that? Honestly I didn't do anything but it ran successfully locally so I just reran the workflow.

**Pull request merged**